### PR TITLE
home-assistant-custom-components.garmin_connect: init at unstable-2024-08-31

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This packages the garmin_connect component from cyberjunky.

I decided to not package the latest release, but the master branch, because the latest release hard depends on a specific version of python3Packages.garminconnect. The master branch does not.

--

Opening this as draft until I confirmed it works (or someone else has).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
This works as expected, except I had to use `python312Packages` instead of `python3Packages` to import the `propagatedBuildInputs`.

@mweinelt what to do about this?

File: pkgs/servers/home-assistant/custom-components/garmin_connect/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  buildHomeAssistantComponent,
+  fetchFromGitHub,
+  python3Packages,
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/garmin_connect/default.nix:5]: You can't mix package sets here.

```suggestion
  garminconnect,
  tzlocal,
```

File: pkgs/servers/home-assistant/custom-components/garmin_connect/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  propagatedBuildInputs = [
+    garminconnect
+    tzlocal
+  ];
+
+  meta = with lib; {
+    description = "The Garmin Connect integration allows you to expose data from Garmin Connect to Home Assistant";
+    homepage = "https://github.com/cyberjunky/home-asstsant-garmin_connect";
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/garmin_connect/default.nix:28]: broken url

Comment (matthiasbeyer) [reply pkgs/servers/home-assistant/custom-components/garmin_connect/default.nix:28]: Good catch!

**Comment by dmadisetti** [comment]
Dupe of https://github.com/NixOS/nixpkgs/pull/300507

But I'm here for it! Will close that in favor of this.
I gave cyberjunky $5, asking for a release but I guess you can't bribe releases :(

**Review by dmadisetti — APPROVED** [review]
@matthiasbeyer feel free to add me as a maintainer

**Comment by mweinelt** [comment]
CI isn't happy with formatting.

**Comment by matthiasbeyer** [comment]
Cleaned up.

**Pull request merged**